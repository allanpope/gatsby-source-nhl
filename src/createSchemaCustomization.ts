import { NodePluginArgs } from 'gatsby';

const createSchemaCustomization = ({ actions }: NodePluginArgs) => {
  const { createTypes } = actions;

  createTypes(`
    type NHLRoster implements Node {
      player: NHLPlayer @link,
      position: NHLPosition @link,
      team: NHLTeam @link
    }

    type NHLPosition implements Node {
      players: [NHLPlayer] @link
    }

    type NHLPlayer implements Node {
      position: NHLPosition @link
      team: NHLTeam @link
    }

    type NHLFranchise implements Node {
      team: NHLTeam @link
    }

    type NHLDivision implements Node {
      teams: [NHLTeam] @link
    }

    type NHLConference implements Node {
      teams: [NHLTeam] @link
    }

    type NHLVenue implements Node {
      team: NHLTeam @link
    }
  `);

  return null;
};

export default createSchemaCustomization;
