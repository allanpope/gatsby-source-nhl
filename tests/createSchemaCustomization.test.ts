import { NodePluginArgs } from 'gatsby';
import createSchemaCustomization from '../src/createSchemaCustomization';

describe('createSchemaCustomization', () => {
  const mockCreateTypes = jest.fn();
  const createNodeHelpers = ({
    actions: {
      createTypes: mockCreateTypes,
    },
  } as unknown) as NodePluginArgs;

  beforeEach(() => {
    createSchemaCustomization(createNodeHelpers);
  });

  it('calls #createTypes with the relational schema', () => {
    expect(mockCreateTypes).toHaveBeenCalledTimes(1);
    expect(mockCreateTypes).toMatchSnapshot();
  });
});
