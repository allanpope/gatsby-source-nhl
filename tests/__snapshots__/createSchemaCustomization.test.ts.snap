// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createSchemaCustomization calls #createTypes with the relational schema 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "
    type NHLRoster implements Node {
      player: NHLPlayer @link,
      position: NHLPosition @link,
      team: NHLTeam @link
    }

    type NHLPosition implements Node {
      players: [NHLPlayer] @link
    }

    type NHLPlayer implements Node {
      position: NHLPosition @link
      team: NHLTeam @link
    }

    type NHLFranchise implements Node {
      team: NHLTeam @link
    }

    type NHLDivision implements Node {
      teams: [NHLTeam] @link
    }

    type NHLConference implements Node {
      teams: [NHLTeam] @link
    }

    type NHLVenue implements Node {
      team: NHLTeam @link
    }

     type NHLTeam implements Node {
      venue: NHLVenue @link
    }
  ",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
